zipserver/archive.go

```diff
@@ -96,61 +147,79 @@ func (a *Archiver) sendZipExtracted(prefix, fname string, limits *ExtractLimits)
 	defer zipReader.Close()
 
 	fileCount := 0
-	byteCount := 0
+	byteCount := uint64(0)
+	uploadedCount := 0
 
 	for _, file := range zipReader.File {
+		if shouldIgnoreFile(file.Name) {
+			continue
+		}
+
 		if len(file.Name) > limits.MaxFileNameLength {
 			return nil, fmt.Errorf("Zip contains file paths that are too long")
 		}
 
 		if file.UncompressedSize64 > uint64(limits.MaxFileSize) {
 			return nil, fmt.Errorf("Zip contains file that is too large (%s)", file.Name)
```

Comment:
A far as we're concerned, why would "100 10MB files" be acceptable, but "one 1GB file" not? It's not like individual items can be larger than the total size of the game anyway :)